wasmtime-wasi/src/host_impls.rs

```diff
@@ -88,10 +88,66 @@ fn convert_errno(error: Errno) -> host::__wasi_errno_t {
     }
 }
 
+fn fd_prestats_get_entry(
+    pt: &host::fd_prestats,
+    fd: host::__wasi_fd_t,
+) -> Option<&host::fd_prestat> {
+    // Test for file descriptor existence
+    if fd as usize >= pt.size {
+        return None;
+    }
+
+    let prestat = unsafe { &*pt.prestats.add(fd as usize) };
+    if prestat.dir == ::std::ptr::null() {
+        return None;
+    }
+
+    Some(prestat)
+}
+
+macro_rules! rwlock_rdlock {
```

Comment:
Given the fact that we'll be using `host::rwlock_rdlock` (and `host::rwlock_unlock`) in a couple more places before we port every single bit of code from C that is using it, I've decided to add these simple macros, just for convenience.